src/gradata/_tenant.py

```
@@ -0,0 +1,93 @@
+    fpath = brain / TENANT_FILE
+    if fpath.exists():
+        tid = fpath.read_text(encoding="utf-8").strip()
+        if _is_valid_uuid(tid):
+            return tid
+
+    brain.mkdir(parents=True, exist_ok=True)
+    tid = str(uuid.uuid4())
+    try:
+        with open(fpath, "x", encoding="utf-8") as fh:
+            fh.write(tid)
+        return tid
```

Ruff (0.15.10)

[warning] Consider moving this statement to an `else` block

(TRY300)